stdlib/ysh/args.ysh

```diff
@@ -80,26 +79,46 @@ proc parser (; place ; ; block_def) {
   call place->setValue(p)
 }
 
-proc flag (short, long ; type='bool' ; default=null, help=null) {
+const kValidTypes = [Bool, Float, Int, Str]
+const kValidTypeNames = []
+for vt in (kValidTypes) {
+  call kValidTypeNames->append(vt.name)
+}
+
+func isValidType (type) {
+  try {
```

Comment:
I think the  `try` is unnecessary here?


---

Actually I think this is another interesting argument for defining `obj1 === obj2` as identity

because then we can replace the entire `isValidType()` function with `mytype in kValidTypes`

Reply:
The `try` is necessary because `is` raises an error on mismatched types:
```
ysh-0.23.0$ = 'int' is Int
  = 'int' is Int
          ^~
[ interactive ]:1: fatal: Mismatched types
```
So if a caller uses a non-type-object value for the type argument to `flag`, they will get a confusing error.

This behavior was actually a little surprising to me. I think there are more circumstances when I would knowingly want to use `is` with potentially mismatched types (e.g., validating input parameters) than circumstances when I would be glad that it caught me accidentally using the wrong type. In other words, I would have expected `false` rather than an error for a mismatched type.

Reply:
Hmmm yeah I think we should remove that check!   At least, it is inconsistent with Python

And JavaScript it seems